include/glow/Optimizer/FunctionPass.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_OPTIMIZER_FUNCTIONPASS_H
+#define GLOW_OPTIMIZER_FUNCTIONPASS_H
+
+namespace glow {
+
+class FunctionPass {
+protected:
+  /// Perform some tasks before the pass.
+  virtual void doInitialization(Function *F) const {
+    LOG_SCOPE(F->getLogContext(), getName());
+  }
+
+  /// Perform some tasks after the pass.
+  virtual void doFinalization(Function *F) const {}
+
+  /// Implementation of the pass. \returns whether the pass modifies \p F.
+  virtual bool passImpl(Function *F) = 0;
```

Comment:
Does it need to be the part of the API? `runFunction` should be enough for the clients. And a pass can decide for itself if it wants to put all of its logic into `runFunction` or if it needs to use a helper.

Reply:
Makes sense, I guess the pass could be responsible for implementing any initialization/finalization logic in `runFunction()`. I can revert moving of `LOG_SCOPE()` and wait until the pass manager lands to put it there.